tests/system/src/test/java/test/javafx/scene/StyleMemoryLeakTest.java

```diff
@@ -0,0 +1,102 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Group;
+import javafx.scene.Node;
+import javafx.scene.Parent;
+import javafx.scene.control.Button;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import junit.framework.Assert;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+import test.util.memory.JMemoryBuddy;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.fail;
+import static org.junit.Assert.assertTrue;
+
+
+public class StyleMemoryLeakTest {
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        CountDownLatch startupLatch = new CountDownLatch(1);
+        Platform.startup(() -> {
+            Platform.setImplicitExit(false);
+            startupLatch.countDown();
+        });
+        assertTrue("Timeout waiting for FX runtime to start", startupLatch.await(15, TimeUnit.SECONDS));
+    }
+
+    @Test
+    public void testRootNodeMemoryLeak() throws Exception {
+        JMemoryBuddy.memoryTest((checker) -> {
+            CountDownLatch showingLatch = new CountDownLatch(1);
+
+            Button toBeRemoved = new Button();
+            Group root = new Group();
+            AtomicReference<Stage> stage = new AtomicReference<>();
+
+            Util.runAndWait(() -> {
+                stage.set(new Stage());
+                stage.get().setOnShown(l -> {
+                    Platform.runLater(() -> showingLatch.countDown());
+                });
+                stage.get().setScene(new Scene(root));
+                stage.get().show();
+            });
+
+            try {
+                assertTrue("Timeout waiting test stage", showingLatch.await(15, TimeUnit.SECONDS));
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            Util.runAndWait(() -> {
+                root.getChildren().clear();
+                stage.get().hide();
+            });
+
+            checker.assertCollectable(stage.get());
+            checker.setAsReferenced(toBeRemoved);
+        });
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Platform.runLater(() -> {
+            Platform.exit();
+        });
```

Comment:
`Platform.exit()` can be called from any thread. So `Platform.runLater` is not required.

Reply:
done